pep-9999.rst

```diff
@@ -0,0 +1,257 @@
+PEP: 9999
+Title: Package index configuration standardization
+Author: Fredrik Averpil <fredrik at averpil.com>
+Status: Draft
+Type: Informational
+Created: 05-May-2021
+Post-History: <dates of postings to python-ideas and/or python-dev>
+
+
+Abstract
+========
+
+This PEP standardizes configuration of PyPI repositories.
+
+
+Motivation
+==========
+
+Workflow tools come up with their own configuration mechanisms for PyPI repositories, as there is no standard for this. This causes fragmentation and pain points.
+
+Examples of such configuration files:
+
+* Pip's ``pip.conf`` and ``pip.ini``
+* Packaging's ``.pypirc`` (also used by tools like Flit and Twine)
+* Poetry's ``config.toml`` and ``auth.toml``
+
+
+Rationale
+=========
+
+While this PEP aim to come up with a unified PyPI repository configuration, thought has been given so to be able to extend the configuration later. For example, PyPI repositories are assumed in this PEP, but a ``type`` attribute could be introduced which would default to ``pypi`` down the line, to allow for other package repository types. Any tool could also start taking advantage of the fact that the configuration files and their discovery is a standard, and use this for their own configuration.
+
+
+Specification
+=============
+
+The specification consists of three parts: the configuration definition, package discovery and configuration file discovery.
+
+
+Configuration definition
+------------------------
+
+The configuration is split into two files:
+
+* ``config.toml`` for PyPI repository configuration
+* ``auth.toml`` for credentials
+
+
+The config.toml file
+''''''''''''''''''''
+
+Package index configuration is defined in the ``config.toml`` file::
+
+
+    [repositories]
+
+    [repostiories.pypi]
+    url = "https://pypi.org/simple/"
+
```

Comment:
```suggestion
    [[repostiories]]
    name = "pypi"
    url = "https://pypi.org/simple/"

```

Should the name `pypi` should be reserved? I honestly do not feel good about being able to change the url for these.

Reply:
There is also the question of mirrors which has come up a few times.

Reply:
I feel great about being able to change it ;) (see my comment above)

But it should certainly be discouraged unless someone really knows what they're doing. That's for documentation, and worth noting here, but not something we should forbid in the spec.

Reply:
> There is also the question of mirrors which has come up a few times.

I would just specify a mirror as another package index (with Poetry). If I wanted to, I could either make it take precedence over PyPI.org (the `priority` attribute) or I could disable PyPI.org alltogether (the `enable` attribute).

> I honestly do not feel good about being able to change the url for these.

The spec would become simpler if we did not address this. But in my opinion, we _could_ reserve the PyPI.org attributes `name` and `url` so that they cannot be modified. I'm just not sure what good that will do, if we allow the `priority` and `enable` attributes on this package index - as then you could re-order or disable PyPI.org anyway.